src/Ship.js

```diff
@@ -0,0 +1,29 @@
+class Ship {
+  constructor(itinerary) {
+    this.itinerary = itinerary;
+    this.currentPort = itinerary.ports[0];
+    this.previousPort = null;
+    this.currentPort.addShip(this);
+  }
+  setSail() {
+    const itinerary = this.itinerary;
+    const currentPortIndex = itinerary.ports.indexOf(this.currentPort);
+
+    this.currentPort.removeShip(this);
+
+    if (currentPortIndex === itinerary.ports.length - 1) {
+      throw new Error("End of itinerary reached");
+    }
+    this.previousPort = this.currentPort;
+    this.currentPort = null;
+  }
+  dock() {
+    const itinerary = this.itinerary;
+    const previousPotIndex = itinerary.ports.indexOf(this.previousPort);
+
+    this.currentPort = itinerary.ports[previousPotIndex + 1];
+    this.currentPort.addShip(this);
```

Comment:
As above, could assign the `ports` to a variable instead of `itinerary` as you only ever use the `itinerary` to access the `ports`

Reply:
Amended like the one above. Test's still passing.